`podman manifest annotate --index` panics with `panic: runtime error: index out of range [1] with length 1`

### Issue Description

Annotate panics when trying to use it.

`podman version`
```text
Client:       Podman Engine
Version:      5.3.1
API Version:  5.3.1
Go Version:   go1.23.3
Built:        Thu Nov 21 00:00:00 2024
OS/Arch:      linux/amd64
```

### Steps to reproduce the issue

Steps to reproduce the issue
1. `podman run --platform linux/amd64 --rm -it quay.io/podman/stable:latest`
2. `podman manifest create asdf:asdf`
3. `podman manifest annotate --index asdf:asdf --annotation asdf=asdf`


### Describe the results you received

```text
panic: runtime error: index out of range [1] with length 1

goroutine 1 gp=0xc0000061c0 m=0 mp=0x5638b07e2f80 [running]:
panic({0x5638af923d20?, 0xc0006300f0?})
        /usr/lib/golang/src/runtime/panic.go:804 +0x168 fp=0xc000053940 sp=0xc000053890 pc=0x5638addb5a88
runtime.goPanicIndex(0x1, 0x1)
        /usr/lib/golang/src/runtime/panic.go:115 +0x74 fp=0xc000053980 sp=0xc000053940 pc=0x5638add79934
github.com/containers/podman/v5/cmd/podman/manifest.annotate(0x5638b0707580?, {0xc00015a380, 0x1, 0x4?})
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/cmd/podman/manifest/annotate.go:122 +0x53f fp=0xc000053bc0 sp=0xc000053980 pc=0x5638af01d73f
github.com/spf13/cobra.(*Command).execute(0x5638b0707580, {0xc000040260, 0x4, 0x4})
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/vendor/github.com/spf13/cobra/command.go:985 +0xaaa fp=0xc000053d48 sp=0xc000053bc0 pc=0x5638ae24b9aa
github.com/spf13/cobra.(*Command).ExecuteC(0x5638b06f9de0)
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/vendor/github.com/spf13/cobra/command.go:1117 +0x3ff fp=0xc000053e20 sp=0xc000053d48 pc=0x5638ae24c27f
github.com/spf13/cobra.(*Command).Execute(...)
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/vendor/github.com/spf13/cobra/command.go:1041
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/vendor/github.com/spf13/cobra/command.go:1034
main.Execute()
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/cmd/podman/root.go:116 +0xb4 fp=0xc000053ea8 sp=0xc000053e20 pc=0x5638af11ed34
main.main()
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/cmd/podman/main.go:61 +0x4b2 fp=0xc000053f50 sp=0xc000053ea8 pc=0x5638af11e492
runtime.main()
        /usr/lib/golang/src/runtime/proc.go:272 +0x29d fp=0xc000053fe0 sp=0xc000053f50 pc=0x5638add7ec1d
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc000053fe8 sp=0xc000053fe0 pc=0x5638addbe541

goroutine 2 gp=0xc000006c40 m=nil [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008afa8 sp=0xc00008af88 pc=0x5638addb5e6e
runtime.goparkunlock(...)
        /usr/lib/golang/src/runtime/proc.go:430
runtime.forcegchelper()
        /usr/lib/golang/src/runtime/proc.go:337 +0xb8 fp=0xc00008afe0 sp=0xc00008afa8 pc=0x5638add7ef58
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008afe8 sp=0xc00008afe0 pc=0x5638addbe541
created by runtime.init.8 in goroutine 1
        /usr/lib/golang/src/runtime/proc.go:325 +0x1a

goroutine 3 gp=0xc000007180 m=nil [GC sweep wait]:
runtime.gopark(0x1?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008b780 sp=0xc00008b760 pc=0x5638addb5e6e
runtime.goparkunlock(...)
        /usr/lib/golang/src/runtime/proc.go:430
runtime.bgsweep(0xc0000b8000)
        /usr/lib/golang/src/runtime/mgcsweep.go:317 +0xdf fp=0xc00008b7c8 sp=0xc00008b780 pc=0x5638add66bff
runtime.gcenable.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:203 +0x25 fp=0xc00008b7e0 sp=0xc00008b7c8 pc=0x5638add5b285
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008b7e8 sp=0xc00008b7e0 pc=0x5638addbe541
created by runtime.gcenable in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:203 +0x66

goroutine 4 gp=0xc000007340 m=nil [GC scavenge wait]:
runtime.gopark(0x10000?, 0x5638af577b80?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008bf78 sp=0xc00008bf58 pc=0x5638addb5e6e
runtime.goparkunlock(...)
        /usr/lib/golang/src/runtime/proc.go:430
runtime.(*scavengerState).park(0x5638b07dd500)
        /usr/lib/golang/src/runtime/mgcscavenge.go:425 +0x49 fp=0xc00008bfa8 sp=0xc00008bf78 pc=0x5638add645e9
runtime.bgscavenge(0xc0000b8000)
        /usr/lib/golang/src/runtime/mgcscavenge.go:658 +0x59 fp=0xc00008bfc8 sp=0xc00008bfa8 pc=0x5638add64b79
runtime.gcenable.gowrap2()
        /usr/lib/golang/src/runtime/mgc.go:204 +0x25 fp=0xc00008bfe0 sp=0xc00008bfc8 pc=0x5638add5b225
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008bfe8 sp=0xc00008bfe0 pc=0x5638addbe541
created by runtime.gcenable in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:204 +0xa5

goroutine 5 gp=0xc000007c00 m=nil [finalizer wait]:
runtime.gopark(0xc00008a648?, 0x5638add50e45?, 0xb0?, 0x1?, 0xc0000061c0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008a620 sp=0xc00008a600 pc=0x5638addb5e6e
runtime.runfinq()
        /usr/lib/golang/src/runtime/mfinal.go:193 +0x107 fp=0xc00008a7e0 sp=0xc00008a620 pc=0x5638add5a307
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008a7e8 sp=0xc00008a7e0 pc=0x5638addbe541
created by runtime.createfing in goroutine 1
        /usr/lib/golang/src/runtime/mfinal.go:163 +0x3d

goroutine 6 gp=0xc00034ec40 m=nil [chan receive]:
runtime.gopark(0xc00008c760?, 0x5638adf5af65?, 0x80?, 0x61?, 0x5638afa53540?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008c718 sp=0xc00008c6f8 pc=0x5638addb5e6e
runtime.chanrecv(0xc0000c6a10, 0x0, 0x1)
        /usr/lib/golang/src/runtime/chan.go:639 +0x41c fp=0xc00008c790 sp=0xc00008c718 pc=0x5638add4a15c
runtime.chanrecv1(0x0?, 0x0?)
        /usr/lib/golang/src/runtime/chan.go:489 +0x12 fp=0xc00008c7b8 sp=0xc00008c790 pc=0x5638add49d12
runtime.unique_runtime_registerUniqueMapCleanup.func1(...)
        /usr/lib/golang/src/runtime/mgc.go:1732
runtime.unique_runtime_registerUniqueMapCleanup.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1735 +0x2f fp=0xc00008c7e0 sp=0xc00008c7b8 pc=0x5638add5e2cf
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008c7e8 sp=0xc00008c7e0 pc=0x5638addbe541
created by unique.runtime_registerUniqueMapCleanup in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1730 +0x96

goroutine 7 gp=0xc00034efc0 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008cf38 sp=0xc00008cf18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00008cfc8 sp=0xc00008cf38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00008cfe0 sp=0xc00008cfc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008cfe8 sp=0xc00008cfe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 18 gp=0xc000504380 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000086738 sp=0xc000086718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc0000867c8 sp=0xc000086738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc0000867e0 sp=0xc0000867c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc0000867e8 sp=0xc0000867e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 34 gp=0xc000104380 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011c738 sp=0xc00011c718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00011c7c8 sp=0xc00011c738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00011c7e0 sp=0xc00011c7c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011c7e8 sp=0xc00011c7e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 8 gp=0xc00034f180 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008d738 sp=0xc00008d718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00008d7c8 sp=0xc00008d738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00008d7e0 sp=0xc00008d7c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008d7e8 sp=0xc00008d7e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 9 gp=0xc00034f340 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00008df38 sp=0xc00008df18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00008dfc8 sp=0xc00008df38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00008dfe0 sp=0xc00008dfc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00008dfe8 sp=0xc00008dfe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 10 gp=0xc00034f500 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000118738 sp=0xc000118718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc0001187c8 sp=0xc000118738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc0001187e0 sp=0xc0001187c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc0001187e8 sp=0xc0001187e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 19 gp=0xc000504540 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000086f38 sp=0xc000086f18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc000086fc8 sp=0xc000086f38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc000086fe0 sp=0xc000086fc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc000086fe8 sp=0xc000086fe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 35 gp=0xc0001048c0 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011cf38 sp=0xc00011cf18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00011cfc8 sp=0xc00011cf38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00011cfe0 sp=0xc00011cfc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011cfe8 sp=0xc00011cfe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 50 gp=0xc000584000 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00058a738 sp=0xc00058a718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00058a7c8 sp=0xc00058a738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00058a7e0 sp=0xc00058a7c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00058a7e8 sp=0xc00058a7e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 11 gp=0xc00034f6c0 m=nil [GC worker (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000118f38 sp=0xc000118f18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc000118fc8 sp=0xc000118f38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc000118fe0 sp=0xc000118fc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc000118fe8 sp=0xc000118fe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 12 gp=0xc00034f880 m=nil [GC worker (idle)]:
runtime.gopark(0x5638b080b500?, 0x1?, 0x19?, 0x3f?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000119738 sp=0xc000119718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc0001197c8 sp=0xc000119738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc0001197e0 sp=0xc0001197c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc0001197e8 sp=0xc0001197e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 13 gp=0xc00034fa40 m=nil [GC worker (idle)]:
runtime.gopark(0x3701c342865b2?, 0x3?, 0xfe?, 0xef?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000119f38 sp=0xc000119f18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc000119fc8 sp=0xc000119f38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc000119fe0 sp=0xc000119fc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc000119fe8 sp=0xc000119fe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 14 gp=0xc00034fc00 m=nil [GC worker (idle)]:
runtime.gopark(0x3701c34286702?, 0x1?, 0x7e?, 0x83?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011a738 sp=0xc00011a718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00011a7c8 sp=0xc00011a738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00011a7e0 sp=0xc00011a7c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011a7e8 sp=0xc00011a7e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 36 gp=0xc000104a80 m=nil [GC worker (idle)]:
runtime.gopark(0x3701c342865b6?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011d738 sp=0xc00011d718 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00011d7c8 sp=0xc00011d738 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00011d7e0 sp=0xc00011d7c8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011d7e8 sp=0xc00011d7e0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 51 gp=0xc0005841c0 m=nil [GC worker (idle)]:
runtime.gopark(0x3701c342886b5?, 0x1?, 0x11?, 0xdb?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00058af38 sp=0xc00058af18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00058afc8 sp=0xc00058af38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00058afe0 sp=0xc00058afc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00058afe8 sp=0xc00058afe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 15 gp=0xc00034fdc0 m=nil [GC worker (idle)]:
runtime.gopark(0x3701c3426558c?, 0x3?, 0xcc?, 0x5e?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011af38 sp=0xc00011af18 pc=0x5638addb5e6e
runtime.gcBgMarkWorker(0xc0000c7810)
        /usr/lib/golang/src/runtime/mgc.go:1363 +0xe9 fp=0xc00011afc8 sp=0xc00011af38 pc=0x5638add5d5c9
runtime.gcBgMarkStartWorkers.gowrap1()
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x25 fp=0xc00011afe0 sp=0xc00011afc8 pc=0x5638add5d4a5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011afe8 sp=0xc00011afe0 pc=0x5638addbe541
created by runtime.gcBgMarkStartWorkers in goroutine 1
        /usr/lib/golang/src/runtime/mgc.go:1279 +0x105

goroutine 37 gp=0xc000685180 m=nil [select]:
runtime.gopark(0xc00011b788?, 0x2?, 0x8?, 0x11?, 0xc00011b784?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00011b628 sp=0xc00011b608 pc=0x5638addb5e6e
runtime.selectgo(0xc00011b788, 0xc00011b780, 0x0?, 0x0, 0x0?, 0x1)
        /usr/lib/golang/src/runtime/select.go:335 +0x7a5 fp=0xc00011b750 sp=0xc00011b628 pc=0x5638add91225
database/sql.(*DB).connectionOpener(0xc0005b5520, {0x5638afa36b90, 0xc000148000})
        /usr/lib/golang/src/database/sql/sql.go:1253 +0x87 fp=0xc00011b7b8 sp=0xc00011b750 pc=0x5638ae5ba0c7
database/sql.OpenDB.gowrap1()
        /usr/lib/golang/src/database/sql/sql.go:833 +0x28 fp=0xc00011b7e0 sp=0xc00011b7b8 pc=0x5638ae5b83e8
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00011b7e8 sp=0xc00011b7e0 pc=0x5638addbe541
created by database/sql.OpenDB in goroutine 1
        /usr/lib/golang/src/database/sql/sql.go:833 +0x130

goroutine 16 gp=0xc000685880 m=nil [select]:
runtime.gopark(0xc000088790?, 0x2?, 0x0?, 0x0?, 0xc00008869c?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000088538 sp=0xc000088518 pc=0x5638addb5e6e
runtime.selectgo(0xc000088790, 0xc000088698, 0x5638add50e45?, 0x0, 0x5638af9aacc0?, 0x1)
        /usr/lib/golang/src/runtime/select.go:335 +0x7a5 fp=0xc000088660 sp=0xc000088538 pc=0x5638add91225
github.com/containers/podman/v5/libpod/shutdown.Start.func1()
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/shutdown/handler.go:56 +0x87 fp=0xc0000887e0 sp=0xc000088660 pc=0x5638aec791a7
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc0000887e8 sp=0xc0000887e0 pc=0x5638addbe541
created by github.com/containers/podman/v5/libpod/shutdown.Start in goroutine 1
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/shutdown/handler.go:55 +0xe5

goroutine 43 gp=0xc000685c00 m=nil [select]:
runtime.gopark(0xc00009ff98?, 0x2?, 0x80?, 0x20?, 0xc00009fe94?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00009fc78 sp=0xc00009fc58 pc=0x5638addb5e6e
runtime.selectgo(0xc00009ff98, 0xc00009fe90, 0x0?, 0x0, 0x0?, 0x1)
        /usr/lib/golang/src/runtime/select.go:335 +0x7a5 fp=0xc00009fda0 sp=0xc00009fc78 pc=0x5638add91225
github.com/containers/podman/v5/libpod.(*Runtime).libimageEvents.func2()
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/runtime.go:746 +0xda fp=0xc00009ffe0 sp=0xc00009fda0 pc=0x5638aee4af5a
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00009ffe8 sp=0xc00009ffe0 pc=0x5638addbe541
created by github.com/containers/podman/v5/libpod.(*Runtime).libimageEvents in goroutine 1
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/runtime.go:719 +0xfc

goroutine 44 gp=0xc000685dc0 m=nil [chan receive]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc000089f08 sp=0xc000089ee8 pc=0x5638addb5e6e
runtime.chanrecv(0xc0002b09a0, 0xc000089fc8, 0x1)
        /usr/lib/golang/src/runtime/chan.go:639 +0x41c fp=0xc000089f80 sp=0xc000089f08 pc=0x5638add4a15c
runtime.chanrecv2(0x0?, 0x0?)
        /usr/lib/golang/src/runtime/chan.go:494 +0x12 fp=0xc000089fa8 sp=0xc000089f80 pc=0x5638add49d32
github.com/containers/podman/v5/libpod.(*Runtime).startWorker.func1()
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/runtime_worker.go:11 +0x6c fp=0xc000089fe0 sp=0xc000089fa8 pc=0x5638aee67e8c
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc000089fe8 sp=0xc000089fe0 pc=0x5638addbe541
created by github.com/containers/podman/v5/libpod.(*Runtime).startWorker in goroutine 1
        /builddir/build/BUILD/podman-5.3.1-build/podman-5.3.1/libpod/runtime_worker.go:10 +0x4f

goroutine 45 gp=0xc000104fc0 m=nil [select, locked to thread]:
runtime.gopark(0xc00058cfa8?, 0x2?, 0x0?, 0x0?, 0xc00058cf94?)
        /usr/lib/golang/src/runtime/proc.go:424 +0xce fp=0xc00058ce30 sp=0xc00058ce10 pc=0x5638addb5e6e
runtime.selectgo(0xc00058cfa8, 0xc00058cf90, 0x0?, 0x0, 0x0?, 0x1)
        /usr/lib/golang/src/runtime/select.go:335 +0x7a5 fp=0xc00058cf58 sp=0xc00058ce30 pc=0x5638add91225
runtime.ensureSigM.func1()
        /usr/lib/golang/src/runtime/signal_unix.go:1077 +0x1a5 fp=0xc00058cfe0 sp=0xc00058cf58 pc=0x5638addacda5
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00058cfe8 sp=0xc00058cfe0 pc=0x5638addbe541
created by runtime.ensureSigM in goroutine 1
        /usr/lib/golang/src/runtime/signal_unix.go:1060 +0xc8

goroutine 46 gp=0xc000105180 m=9 mp=0xc000600008 [syscall]:
runtime.notetsleepg(0x5638b080a440, 0xffffffffffffffff)
        /usr/lib/golang/src/runtime/lock_futex.go:246 +0x29 fp=0xc00058d7a0 sp=0xc00058d778 pc=0x5638add4fd69
os/signal.signal_recv()
        /usr/lib/golang/src/runtime/sigqueue.go:152 +0x29 fp=0xc00058d7c0 sp=0xc00058d7a0 pc=0x5638addb7f29
os/signal.loop()
        /usr/lib/golang/src/os/signal/signal_unix.go:23 +0x13 fp=0xc00058d7e0 sp=0xc00058d7c0 pc=0x5638adeab7f3
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc00058d7e8 sp=0xc00058d7e0 pc=0x5638addbe541
created by os/signal.Notify.func1.1 in goroutine 1
        /usr/lib/golang/src/os/signal/signal.go:151 +0x1f
```

### Describe the results you expected

Expecting the call to succeed

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.38.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.12-3.fc41.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: '
  cpuUtilization:
    idlePercent: 99.98
    systemPercent: 0.01
    userPercent: 0.01
  cpus: 16
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "41"
  eventLogger: file
  freeLocks: 2048
  hostname: 649fe360b7d4
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.15.167.4-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 6351409152
  memTotal: 7908708352
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.13.1-1.fc41.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.13.1
    package: netavark-1.13.0-1.fc41.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.13.0
  ociRuntime:
    name: crun
    package: crun-1.18.1-1.fc41.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.18.1
      commit: c41f034fdbb9742c395085fc98459c94ad1f9aae
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20241121.g238c69f-1.fc41.x86_64
    version: |
      pasta 0^20241121.g238c69f-1.fc41.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 2147483648
  swapTotal: 2147483648
  uptime: 268h 49m 10.00s (Approximately 11.17 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /usr/lib/containers/storage
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.13-2.fc41.x86_64
      Version: |-
        fusermount3 version: 3.16.2
        fuse-overlayfs: version 1.13-dev
        FUSE library version 3.16.2
        using FUSE kernel interface version 7.38
    overlay.mountopt: nodev,fsync=0
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 32891523072
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 5.3.1
  Built: 1732147200
  BuiltTime: Thu Nov 21 00:00:00 2024
  GitCommit: ""
  GoVersion: go1.23.3
  Os: linux
  OsArch: linux/amd64
  Version: 5.3.1
```

### Podman in a container

Yes

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

N/A

### Additional information

Workaround: `podman manifest create asdf:asdf --annotation asdf=asdf`

Think its this:
https://github.com/containers/podman/blob/bf1661c93027cc723473ead3746585289190efc1/cmd/podman/manifest/annotate.go#L122

But i can't specify a second arg because this line stops me:
https://github.com/containers/podman/blob/bf1661c93027cc723473ead3746585289190efc1/cmd/podman/manifest/annotate.go#L95-L97

--- a/cmd/podman/manifest/annotate.go
+++ b/cmd/podman/manifest/annotate.go
@@ -119,7 +119,7 @@ func annotate(cmd *cobra.Command, args []string) error {
 	} else {
 		opts.Annotations = annotations
 	}
-	id, err := registry.ImageEngine().ManifestAnnotate(registry.Context(), args[0], args[1], opts)
+	id, err := registry.ImageEngine().ManifestAnnotate(registry.Context(), listImageSpec, instanceSpec, opts)
 	if err != nil {
 		return err
 	}

--- a/pkg/api/handlers/libpod/manifests.go
+++ b/pkg/api/handlers/libpod/manifests.go
@@ -706,14 +706,20 @@ func ManifestModify(w http.ResponseWriter, r *http.Request) {
 		}
 	case strings.EqualFold("annotate", body.Operation):
 		options := body.ManifestAnnotateOptions
-		for _, image := range body.Images {
+		images := []string{""}
+		if len(body.Images) > 0 {
+			images = body.Images
+		}
+		for _, image := range images {
 			id, err := imageEngine.ManifestAnnotate(r.Context(), name, image, options)
 			if err != nil {
 				report.Errors = append(report.Errors, err)
 				continue
 			}
 			report.ID = id
-			report.Images = append(report.Images, image)
+			if image != "" {
+				report.Images = append(report.Images, image)
+			}
 		}
 	default:
 		utils.Error(w, http.StatusBadRequest, fmt.Errorf("illegal operation %q for %q", body.Operation, r.URL.String()))

--- a/pkg/bindings/manifests/types.go
+++ b/pkg/bindings/manifests/types.go
@@ -19,7 +19,7 @@ type InspectOptions struct {
 type CreateOptions struct {
 	All        *bool
 	Amend      *bool
-	Annotation map[string]string
+	Annotation map[string]string `json:"annotations" schema:"annotations"`
 }
 
 // ExistsOptions are optional options for checking
@@ -35,7 +35,7 @@ type ExistsOptions struct {
 type AddOptions struct {
 	All *bool
 
-	Annotation map[string]string
+	Annotation map[string]string `json:"annotations" schema:"annotations"`
 	Arch       *string
 	Features   []string
 	OS         *string
@@ -54,7 +54,7 @@ type AddOptions struct {
 //
 //go:generate go run ../generator/generator.go AddArtifactOptions
 type AddArtifactOptions struct {
-	Annotation map[string]string
+	Annotation map[string]string `json:"annotations" schema:"annotations"`
 	Arch       *string
 	Features   []string
 	OS         *string
@@ -87,13 +87,14 @@ type ModifyOptions struct {
 	Operation *string
 	All       *bool // All when true, operate on all images in a manifest list that may be included in Images
 
-	Annotations map[string]string // Annotations to add to the entries for Images in the manifest list
-	Arch        *string           // Arch overrides the architecture for the image
-	Features    []string          // Feature list for the image
-	OS          *string           // OS overrides the operating system for the image
-	OSFeatures  []string          `json:"os_features" schema:"os_features"` // OSFeatures overrides the OS features for the image
-	OSVersion   *string           `json:"os_version" schema:"os_version"`   // OSVersion overrides the operating system version for the image
-	Variant     *string           // Variant overrides the architecture variant for the image
+	Annotations      map[string]string // Annotations to add to the entries for Images in the manifest list
+	IndexAnnotations map[string]string `json:"index_annotations" schema:"index_annotations"` // Annotations to add to the manifest list as a whole
+	Arch             *string           // Arch overrides the architecture for the image
+	Features         []string          // Feature list for the image
+	OS               *string           // OS overrides the operating system for the image
+	OSFeatures       []string          `json:"os_features" schema:"os_features"` // OSFeatures overrides the OS features for the image
+	OSVersion        *string           `json:"os_version" schema:"os_version"`   // OSVersion overrides the operating system version for the image
+	Variant          *string           // Variant overrides the architecture variant for the image
 
 	Images        []string // Images is an optional list of images to add/remove to/from manifest list depending on operation
 	Authfile      *string

--- a/pkg/bindings/manifests/types_modify_options.go
+++ b/pkg/bindings/manifests/types_modify_options.go
@@ -63,6 +63,21 @@ func (o *ModifyOptions) GetAnnotations() map[string]string {
 	return o.Annotations
 }
 
+// WithIndexAnnotations set annotations to add to the manifest list as a whole
+func (o *ModifyOptions) WithIndexAnnotations(value map[string]string) *ModifyOptions {
+	o.IndexAnnotations = value
+	return o
+}
+
+// GetIndexAnnotations returns value of annotations to add to the manifest list as a whole
+func (o *ModifyOptions) GetIndexAnnotations() map[string]string {
+	if o.IndexAnnotations == nil {
+		var z map[string]string
+		return z
+	}
+	return o.IndexAnnotations
+}
+
 // WithArch set arch overrides the architecture for the image
 func (o *ModifyOptions) WithArch(value string) *ModifyOptions {
 	o.Arch = &value

--- a/pkg/domain/entities/manifest.go
+++ b/pkg/domain/entities/manifest.go
@@ -82,9 +82,9 @@ type ManifestAnnotateOptions struct {
 	// Variant for the item in the manifest list
 	Variant string `json:"variant" schema:"variant"`
 	// IndexAnnotation is a slice of key=value annotations to add to the manifest list itself
-	IndexAnnotation []string `json:"index_annotation" schema:"annotation"`
+	IndexAnnotation []string `json:"index_annotation" schema:"index_annotation"`
 	// IndexAnnotations is a map of key:value annotations to add to the manifest list itself, by a map which is preferred over IndexAnnotation
-	IndexAnnotations map[string]string `json:"index_annotations" schema:"annotations"`
+	IndexAnnotations map[string]string `json:"index_annotations" schema:"index_annotations"`
 	// IndexSubject is a subject value to set in the manifest list itself
 	IndexSubject string `json:"subject" schema:"subject"`
 }

--- a/pkg/domain/infra/abi/manifest.go
+++ b/pkg/domain/infra/abi/manifest.go
@@ -49,9 +49,10 @@ func (ir *ImageEngine) ManifestCreate(ctx context.Context, name string, images [
 		}
 	}
 
-	annotateOptions := &libimage.ManifestListAnnotateOptions{}
 	if len(opts.Annotations) != 0 {
-		annotateOptions.IndexAnnotations = opts.Annotations
+		annotateOptions := &libimage.ManifestListAnnotateOptions{
+			IndexAnnotations: opts.Annotations,
+		}
 		if err := manifestList.AnnotateInstance("", annotateOptions); err != nil {
 			return "", err
 		}
@@ -230,18 +231,14 @@ func (ir *ImageEngine) ManifestAdd(ctx context.Context, name string, images []st
 			Variant:      opts.Variant,
 			Subject:      opts.IndexSubject,
 		}
-		if len(opts.Annotation) != 0 {
-			annotations := make(map[string]string)
-			for _, annotationSpec := range opts.Annotation {
-				key, val, hasVal := strings.Cut(annotationSpec, "=")
-				if !hasVal {
-					return "", fmt.Errorf("no value given for annotation %q", key)
-				}
-				annotations[key] = val
-			}
-			opts.Annotations = envLib.Join(opts.Annotations, annotations)
+
+		if annotateOptions.Annotations, err = mergeAnnotations(opts.Annotations, opts.Annotation); err != nil {
+			return "", err
 		}
-		annotateOptions.Annotations = opts.Annotations
+
+		if annotateOptions.IndexAnnotations, err = mergeAnnotations(opts.IndexAnnotations, opts.IndexAnnotation); err != nil {
+			return "", err
+		}
 
 		if err := manifestList.AnnotateInstance(instanceDigest, annotateOptions); err != nil {
 			return "", err
@@ -380,10 +377,12 @@ func (ir *ImageEngine) ManifestAddArtifact(ctx context.Context, name string, fil
 		Variant:      opts.Variant,
 		Subject:      opts.IndexSubject,
 	}
-	if annotateOptions.Annotations, err = mergeAnnotations(opts.Annotations, opts.Annotation); err != nil {
+
+	if annotateOptions.Annotations, err = mergeAnnotations(opts.ManifestAnnotateOptions.Annotations, opts.ManifestAnnotateOptions.Annotation); err != nil {
 		return "", err
 	}
-	if annotateOptions.IndexAnnotations, err = mergeAnnotations(opts.IndexAnnotations, opts.IndexAnnotation); err != nil {
+
+	if annotateOptions.IndexAnnotations, err = mergeAnnotations(opts.ManifestAnnotateOptions.IndexAnnotations, opts.ManifestAnnotateOptions.IndexAnnotation); err != nil {
 		return "", err
 	}
 

--- a/pkg/domain/infra/tunnel/manifest.go
+++ b/pkg/domain/infra/tunnel/manifest.go
@@ -131,24 +131,41 @@ func (ir *ImageEngine) ManifestAnnotate(ctx context.Context, name, images string
 	options := new(manifests.ModifyOptions).WithArch(opts.Arch).WithVariant(opts.Variant)
 	options.WithFeatures(opts.Features).WithOS(opts.OS).WithOSVersion(opts.OSVersion)
 
-	if len(opts.Annotation) != 0 {
-		annotations := make(map[string]string)
-		for _, annotationSpec := range opts.Annotation {
-			key, val, hasVal := strings.Cut(annotationSpec, "=")
-			if !hasVal {
-				return "", fmt.Errorf("no value given for annotation %q", key)
-			}
-			annotations[key] = val
-		}
-		opts.Annotations = envLib.Join(opts.Annotations, annotations)
+	annotations, err := mergeAnnotations(opts.Annotations, opts.Annotation)
+	if err != nil {
+		return "", err
 	}
-	options.WithAnnotations(opts.Annotations)
+	options.WithAnnotations(annotations)
+
+	indexAnnotations, err := mergeAnnotations(opts.IndexAnnotations, opts.IndexAnnotation)
+	if err != nil {
+		return "", err
+	}
+	options.WithIndexAnnotations(indexAnnotations)
 
 	id, err := manifests.Annotate(ir.ClientCtx, name, []string{images}, options)
 	if err != nil {
 		return id, fmt.Errorf("annotating to manifest list %s: %w", name, err)
 	}
 	return id, nil
+}
+
+func mergeAnnotations(preferred map[string]string, aux []string) (map[string]string, error) {
+	if len(aux) != 0 {
+		auxAnnotations := make(map[string]string)
+		for _, annotationSpec := range aux {
+			key, val, hasVal := strings.Cut(annotationSpec, "=")
+			if !hasVal {
+				return nil, fmt.Errorf("no value given for annotation %q", key)
+			}
+			auxAnnotations[key] = val
+		}
+		if preferred == nil {
+			preferred = make(map[string]string)
+		}
+		preferred = envLib.Join(auxAnnotations, preferred)
+	}
+	return preferred, nil
 }
 
 // ManifestRemoveDigest removes the digest from manifest list

--- a/test/e2e/manifest_test.go
+++ b/test/e2e/manifest_test.go
@@ -10,12 +10,16 @@ import (
 	"strings"
 
 	"github.com/containers/common/libimage/define"
+	"github.com/containers/image/v5/docker/reference"
+	manifest "github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/transports/alltransports"
 	podmanRegistry "github.com/containers/podman/v5/hack/podman-registry-go"
 	. "github.com/containers/podman/v5/test/utils"
 	"github.com/containers/storage/pkg/archive"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gexec"
+	imgspec "github.com/opencontainers/image-spec/specs-go"
 	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
@@ -368,7 +372,7 @@ add_compression = ["zstd"]`), 0o644)
 	})
 
 	It("annotate", func() {
-		session := podmanTest.Podman([]string{"manifest", "create", "foo"})
+		session := podmanTest.Podman([]string{"manifest", "create", "--annotation", "up=down", "foo"})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
 		session = podmanTest.Podman([]string{"manifest", "add", "foo", imageListInstance})
@@ -377,12 +381,50 @@ add_compression = ["zstd"]`), 0o644)
 		session = podmanTest.Podman([]string{"manifest", "annotate", "--annotation", "hello=world,withcomma", "--arch", "bar", "foo", imageListARM64InstanceDigest})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
+		session = podmanTest.Podman([]string{"manifest", "annotate", "--index", "--annotation", "top=bottom", "foo"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
 		session = podmanTest.Podman([]string{"manifest", "inspect", "foo"})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
-		Expect(session.OutputToString()).To(ContainSubstring(`"architecture": "bar"`))
-		// Check added annotation
-		Expect(session.OutputToString()).To(ContainSubstring(`"hello": "world,withcomma"`))
+		// Extract the digest from the name of the image that we just added to the list
+		ref, err := alltransports.ParseImageName(imageListInstance)
+		Expect(err).ToNot(HaveOccurred())
+		dockerReference := ref.DockerReference()
+		referenceWithDigest, ok := dockerReference.(reference.Canonical)
+		Expect(ok).To(BeTrueBecause("we started with a canonical reference"))
+		// Check that the index has all of the information we've just added to it
+		encoded, err := json.Marshal(&imgspecv1.Index{
+			Versioned: imgspec.Versioned{
+				SchemaVersion: 2,
+			},
+			// media type forced because we need to be able to represent annotations
+			MediaType: imgspecv1.MediaTypeImageIndex,
+			Manifests: []imgspecv1.Descriptor{
+				{
+					// from imageListInstance
+					MediaType: manifest.DockerV2Schema2MediaType,
+					Digest:    referenceWithDigest.Digest(),
+					Size:      527,
+					// OS from imageListInstance(?), Architecture/Variant as above
+					Platform: &imgspecv1.Platform{
+						Architecture: "bar",
+						OS:           "linux",
+					},
+					// added above
+					Annotations: map[string]string{
+						"hello": "world,withcomma",
+					},
+				},
+			},
+			// added above
+			Annotations: map[string]string{
+				"top": "bottom",
+				"up":  "down",
+			},
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session.OutputToString()).To(MatchJSON(encoded))
 	})
 
 	It("remove digest", func() {

